### PR TITLE
Upgrade to Django 2.2

### PR DESCRIPTION
**Pull request by ffont**
**Issue(s)**
https://github.com/MTG/freesound/issues/1660

**Description**
This PR upgrades Django to version 2.2. All the steps indicated in https://github.com/MTG/freesound/issues/1660 were followed and tests are passing.

Pending tasks:

- [x] Upgrade django dependencies to versions that will run both on 2.2 and 3.2
- [x] Use `path` instead of `url` in url configuration and use new path syntax where possible (not required but nice improvement) - only partially did that, many apps have not been updated yet
- [x] Read 2.2 release notes
- [ ] Deploy!

**Deployment steps**:
Apply migrations on production. Should not take long.

### CODE REVIEW

File: apiv2/oauth2_urls.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def _wrapped_view_func(request, *args, **kwargs):
     return _wrapped_view_func
 
 
+app_name = 'oauth2_provider'
```
Comment (Bomme) [inline apiv2/oauth2_urls.py:49]: Why is this required?

File: sounds/templatetags/display_sound.py [hunk 2]
```diff
@@ -114,10 +114,10 @@ def sound_object_retrieved_using_bulk_query_id(sound):
             'request': request,
             'is_explicit': sound_obj.is_explicit and
                            (not request.user.is_authenticated or not request.user.profile.is_adult),
-            'is_authenticated': request.user.is_authenticated(),
+            'is_authenticated': request.user.is_authenticated == True,
```
Comment (Bomme) [inline sounds/templatetags/display_sound.py:117]: If this is a `bool` we do not need to compare to `True`

File: apiv2/oauth2_urls.py (hunk 1 above)
Comment (ffont) [reply apiv2/oauth2_urls.py:49]: This was to fix one of the deprecation warnings to update to 2.x. When URLs are included using `include`, the included URLs file now requires the `app_name` to be present. See some info: https://docs.djangoproject.com/en/4.1/topics/http/urls/#url-namespaces-and-included-urlconfs

File: sounds/templatetags/display_sound.py (hunk 2 above)
Comment (ffont) [reply sounds/templatetags/display_sound.py:117]: This is quite interesting. Some cache-related tests were failing because of that, and what I saw is that is_authenticated was not really a `bool` but a cached bool, and that would make tests fail because `is_authenticated` is used to construct cache keys. So I need the comparison to make this a normal `bool`.

However, now I don't see these errors anymore, even if removing the comparison. It is quite strange because I swear I those tests were failing and it actually took quite a long time for me to track down the issue :S
